src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java

```diff
@@ -663,6 +649,66 @@ private StartWorkflowExecutionResponse startWorkflowExecution(
     }
   }
 
+  @Override
+  public StartWorkflowExecutionAsyncResponse StartWorkflowExecutionAsync(
+      StartWorkflowExecutionAsyncRequest startAsyncRequest) throws TException {
+    return measureRemoteCall(
+        ServiceMethod.START_WORKFLOW_EXECUTION_ASYNC,
+        () -> startWorkflowExecutionAsync(startAsyncRequest));
+  }
+
+  private StartWorkflowExecutionAsyncResponse startWorkflowExecutionAsync(
+      StartWorkflowExecutionAsyncRequest startAsyncRequest) throws TException {
+    ThriftResponse<WorkflowService.StartWorkflowExecutionAsync_result> response = null;
+    try {
+      initializeStartWorkflowRequest(startAsyncRequest.getRequest());
+
+      ThriftRequest<WorkflowService.StartWorkflowExecutionAsync_args> request =
+          buildThriftRequest(
+              "StartWorkflowExecutionAsync",
+              new WorkflowService.StartWorkflowExecutionAsync_args(startAsyncRequest));
+
+      response = doRemoteCall(request);
+      WorkflowService.StartWorkflowExecutionAsync_result result =
+          response.getBody(WorkflowService.StartWorkflowExecutionAsync_result.class);
+      if (response.getResponseCode() == ResponseCode.OK) {
+        return result.getSuccess();
+      }
+      if (result.isSetBadRequestError()) {
+        throw result.getBadRequestError();
+      }
+      if (result.isSetSessionAlreadyExistError()) {
+        throw result.getSessionAlreadyExistError();
+      }
+      if (result.isSetServiceBusyError()) {
+        throw result.getServiceBusyError();
+      }
+      if (result.isSetDomainNotActiveError()) {
+        throw result.getDomainNotActiveError();
+      }
+      if (result.isSetLimitExceededError()) {
+        throw result.getLimitExceededError();
+      }
+      if (result.isSetEntityNotExistError()) {
+        throw result.getEntityNotExistError();
+      }
+      throw new TException("StartWorkflowExecution failed with unknown error:" + result);
```

Comment:
missing check on clientversion not supported error. This bug existed in the original sync apis

Reply:
Fixed. I thought this might be a new error that none of the endpoints accounted for but it seems that StartWorkflowExecution is actually the only one missing it. Nice catch.